joblib/test/test_pickle.py

```diff
@@ -0,0 +1,21 @@
+"""
+Test that pickle and dill output the same bytes
+"""
+import pickle
+
+import nose
+
+from joblib.test.common import dill, with_dill, with_py3
+
+
+def global_func():
+    pass
+
+
+@with_py3
+@with_dill
+def test_dumps_eq():
+    # Test that dill and pickle produce the same results
```

Comment:
I think what we would rather test is that, for picklable objects, joblib pickles don't change whether you use dill or pickle behind the scenes.

Not sure how to test this best. Maybe we would need a way to switch whether dill or pickle is used behind the scenes for all joblib modules at runtime (rather than import time).